connector-email/src/main/java/com/redhat/cloud/notifications/connector/email/EmailRouteBuilder.java

```diff
@@ -111,12 +115,29 @@ protected HttpEndpointBuilderFactory.HttpEndpointBuilder setUpBOPEndpoint() {
         }
     }
 
-    private HttpEndpointBuilderFactory.HttpEndpointBuilder setupRecipientResolverEndpoint() {
+    private HttpEndpointBuilder setupRecipientResolverEndpoint() {
         final String fullURL = emailConnectorConfig.getRecipientsResolverServiceURL() + "/internal/recipients-resolver";
+
         if (fullURL.startsWith("https")) {
-            return https(fullURL.replace("https://", ""))
-                    .sslContextParameters(getSslContextParameters())
-                    .x509HostnameVerifier(NoopHostnameVerifier.INSTANCE);
+            HttpEndpointBuilder endpointBuilder = https(fullURL.replace("https://", ""));
+            if (emailConnectorConfig.getRecipientsResolverTrustStorePath().isPresent() && emailConnectorConfig.getRecipientsResolverTrustStorePassword().isPresent() && emailConnectorConfig.getRecipientsResolverTrustStoreType().isPresent()) {
+
+                KeyStoreParameters keyStoreParameters = new KeyStoreParameters();
+                keyStoreParameters.setResource(emailConnectorConfig.getRecipientsResolverTrustStorePath().get());
+                keyStoreParameters.setPassword(emailConnectorConfig.getRecipientsResolverTrustStorePassword().get());
+                keyStoreParameters.setType(emailConnectorConfig.getRecipientsResolverTrustStoreType().get());
+
+                TrustManagersParameters trustManagersParameters = new TrustManagersParameters();
+                trustManagersParameters.setKeyStore(keyStoreParameters);
+
+                SSLContextParameters sslContextParameters = new SSLContextParameters();
+                sslContextParameters.setTrustManagers(trustManagersParameters);
+
+                endpointBuilder.sslContextParameters(sslContextParameters);
+            } else {
+                Log.warn("TLS is enabled for recipients-resolver but the trust store could not be used to build the Camel endpoint because the trust store path, password or type are missing in the Clowder configuration");
```

Comment:
Shouldn't we invalidate the health check it this error happends ?
My consern is if something goes wrong, all pods could be deployed with an inconsistent ssl config and all requests to the recipients-resolver will probably fail, involving a production outage ?